ui/src/main/java/com/doordeck/sdk/common/models/JWTHeader.java

```diff
@@ -1,222 +1,63 @@
 package com.doordeck.sdk.common.models;
 
 import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.google.common.base.Optional;
+
+import org.immutables.value.Value;
 
 import java.net.URI;
+import java.sql.Time;
 import java.util.Locale;
 import java.util.TimeZone;
 
 //  JWT header
+@Value.Immutable
+@JsonSerialize(as = ImmutableJWTHeader.class)
+@JsonDeserialize(as = ImmutableJWTHeader.class)
 @JsonInclude(JsonInclude.Include.NON_NULL)
```

Comment:
Add @JsonIgnoreProperties(ignoreUnknown=true), this way if they add something you don't expect, it won't fail (it also means you can remove any fields you don't care about without causing an error)

Reply:
Actually yeah, remove anything you don't care about to make this class cleaner